src/mockServiceWorker.js

```diff
@@ -82,22 +82,26 @@ self.addEventListener('message', async function (event) {
   }
 })
 
-async function resolveClient(event) {
+async function resolveMasterClient(event) {
   const client = await self.clients.get(event.clientId)
+
+  if (client.frameType === 'top-level') {
+    return client
+  }
+
   const allClients = await self.clients.matchAll()
-  return client.frameType === 'top-level'
-    ? client
-    : allClients.find((parentClient) => {
-        return (
-          parentClient.frameType === 'top-level' &&
-          parentClient.visibilityState === 'visible' &&
-          client.url.startsWith(parentClient.url)
-        )
-      })
+
+  return allClients
+    .filter((client) => {
+      return client.visibilityState === 'visible'
```

Comment:
By filtering only the visible clients we get a list of clients on the current page. This way we omit all the other possibly open clients.